Makefile.tools

```diff
@@ -0,0 +1,289 @@
+
+$(OBJ_TOOLS_BIN_PCAD_INFER_DECOUPLED) \
+$(OBJ_TOOLS_BIN_PCAD_MACRO_COMPILER) \
+$(OBJ_TOOLS_BIN_PCAD_LIST_MACROS): \
+		$(OBJ_TOOLS_BIN_DIR)/pcad/stamp
+	touch $@
+
+$(OBJ_TOOLS_BIN_DIR)/pcad/stamp: $(OBJ_TOOLS_SRC_DIR)/pcad/Makefile $(CMD_GXX)
+	MAKEFLAGS="$(TOOLS_MAKEFLAGS)" $(SCHEDULER_CMD) --make -- $(MAKE) -C $(dir $<) install CXX=$(abspath $(CMD_GXX))
+	@date > $@
+
+$(OBJ_TOOLS_SRC_DIR)/pcad/Makefile: $(OBJ_TOOLS_SRC_DIR)/pcad/stamp \
+				    $(CMD_PATH_TCLAP_PC) \
+				    $(CMD_PATH_PSON_PC) \
+				    $(CMD_PCONFIGURE) $(CMD_PPKGCONFIG) $(CMD_PHC)
+	mkdir -p $(dir $@)
+	echo 'PREFIX = $(abspath $(OBJ_TOOLS_BIN_DIR))/pcad' >> $(dir $@)/Configfile.local
+	echo 'LANGUAGES += c++' >> $(dir $@)/Configfile.local
+	echo 'LINKOPTS += -Wl,-rpath,$(abspath $(CMD_PATH_GCC_LIB64))' >> $(dir $@)/Configfile.local
+	echo 'COMPILEOPTS += -O3 -march=native -g' >> $(dir $@)/Configfile.local
+	echo 'LINKOPTS += -O3 -march=native -g' >> $(dir $@)/Configfile.local
+	cd $(dir $@); PKG_CONFIG_PATH="$$PKG_CONFIG_PATH:$(abspath $(dir $(CMD_PATH_TCLAP_PC))):$(abspath $(dir $(CMD_PATH_PSON_PC)))" $(abspath $(CMD_PCONFIGURE)) --ppkg-config $(abspath $(CMD_PPKGCONFIG)) --phc $(abspath $(CMD_PHC))
+
+$(OBJ_TOOLS_SRC_DIR)/pcad/stamp: $(shell find src/tools/pcad -type f)
+	mkdir -p $(dir $@)
+	rsync -a --delete src/tools/pcad/ $(OBJ_TOOLS_SRC_DIR)/pcad
+	touch $@
+
+##############################################################################
+# SBT
+##############################################################################
+
+# _plsi_ contrib sbt targets = OBJ_TOOLS_BIN_SBT
+
+OBJ_TOOLS_BIN_SBT = $(OBJ_TOOLS_BIN_DIR)/sbt/sbt
+
+# Caches the sbt binary to the tools directory.
+$(OBJ_TOOLS_BIN_SBT): src/tools/sbt/sbt-launch.jar src/tools/sbt/sbt
+	mkdir -p $(dir $@)
+	cp src/tools/sbt/sbt-launch.jar $(dir $@)
+	cp src/tools/sbt/sbt $(dir $@)
+	chmod +x $@
+
+##############################################################################
+# PSON
+##############################################################################
+
+# _plsi_ contrib pson targets = OBJ_TOOLS_BIN_PSON2JSON OBJ_TOOLS_BIN_PATH_PSON_PC
+
+OBJ_TOOLS_BIN_PSON2JSON = $(OBJ_TOOLS_BIN_DIR)/pson/bin/pson2json
+OBJ_TOOLS_BIN_PATH_PSON_PC = $(OBJ_TOOLS_BIN_DIR)/pson/lib/pkgconfig/pson.pc
+
+# PSON is a C++ JSON parsing library that also allows for JSON-like files that
+# have extra trailing commas floating around.  Piping through this allows
+# stateless JSON emission from various other parts of the code.
+$(OBJ_TOOLS_BIN_PSON2JSON) $(OBJ_TOOLS_BIN_PATH_PSON_PC): $(OBJ_TOOLS_BIN_DIR)/pson/stamp
+	touch $@
+
+$(OBJ_TOOLS_BIN_DIR)/pson/stamp: $(OBJ_TOOLS_SRC_DIR)/pson/Makefile $(CMD_GCC) $(CMD_GXX)
+	MAKEFLAGS="$(TOOLS_MAKEFLAGS)" $(SCHEDULER_CMD) --make -- $(MAKE) -C $(dir $<) CC=$(abspath $(CMD_GCC)) CXX=$(abspath $(CMD_GXX)) install
+	date > $@
+
+$(OBJ_TOOLS_SRC_DIR)/pson/Makefile: \
+		$(OBJ_TOOLS_SRC_DIR)/pson/Configfile \
+		$(CMD_PATH_TCLAP_PC) \
```

Comment:
This is actually intentional. (Apologies if my documentation was not clear; it's at the top of this file.) CMD_PATH_TCLAP_PC is the overridable version (see Makefile.plsi) in case you want to use a prebuilt TCLAP (to save time when running a newly cloned plsi).